src/app/detail/[task_id]/[user_type]/page.tsx

```diff
@@ -83,7 +84,9 @@ export default async function TaskDetailPage({
 
   console.info(`app/detail/${task_id}/${user_type}/page.tsx | Serving user ${token} with payload`, tokenPayload)
 
-  redirectIfResourceNotFound(searchParams, task, !!tokenPayload.internalUserId)
+  if (!!!task) {
+    return <DeletedTaskRedirectPage userType={tokenPayload.internalUserId ? UserRole.IU : UserRole.Client} />
+  }
```

Comment:
Why aren't we passing token as a prop directly here and instead taking a round trip to the task board selector? I think there were 2-3 issues Neil was looking over that was related to this.

Also it's just much more efficient since we are calling the DeletedTaskRedirect component from a server component directly

Reply:
Fixed in 1b1bdd22b0f2b8e4847a14ea55319b86945f2fc0152cc13b43d57b1f3978d6c5cf00373fd4c5e70689a39d18aa04914f69695a181b726fbbecabbbd79a3fbd8da7c940c35c321774deddd92ae6e0e7785b040c81c307cd4c50a7a697ec144ac6b4b6eb2257e4a8853fff900138a8d279 commit